Firestore/Swift/Tests/Integration/CodableIntegrationTests.swift

```diff
@@ -168,6 +169,30 @@ class CodableIntegrationTests: FSTIntegrationTestCase {
     }
   }
 
+  func testAutoPopulatedDocumentId() throws {
+    struct Model: Codable, Equatable {
+      var name: String
+      var docId: AutoPopulatedDocumentId
+    }
+
+    let docToWrite = documentRef()
+    let model = Model(
+      name: "name",
+      docId: AutoPopulatedDocumentId(from: docToWrite)
```

Comment:
When writing, users shouldn't need to set this value to an actual document reference.